packages/flutter_tools/templates/app/linux.tmpl/my_application.cc.tmpl

```diff
@@ -33,8 +36,38 @@ static void my_application_activate(GApplication* application) {
   gtk_widget_grab_focus(GTK_WIDGET(view));
 }
 
+// Implements GApplication::local_command_line.
+static gboolean my_application_local_command_line(GApplication* application, gchar ***arguments, int *exit_status) {
+  MyApplication* self = MY_APPLICATION(application);
+  char** entrypoint_arguments = *arguments;
+  // Strip out the first argument as it is the binary name
+  entrypoint_arguments++;
+  self->dart_entrypoint_arguments = g_strdupv(entrypoint_arguments);
```

Comment:
This would probably be simpler as `self->dart_entrypoint_arguments = g_strdupv(*arguments + 1);`